server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -975,7 +981,23 @@ private void loadFile(File file, OperateFilePlan plan) throws QueryProcessExcept
         createSchemaAutomatically(chunkGroupMetadataList, schemaMap, plan.getSgLevel());
       }
 
-      StorageEngine.getInstance().loadNewTsFile(tsFileResource);
+      List<TsFileResource> splitResources = new ArrayList();
+      try {
+        tsFileResource.getTimePartitionWithCheck();
+      } catch (PartitionViolationException e) {
+        logger.info("try to split the tsfile={}", tsFileResource.getTsFile().getPath());
+        TsFileRewriteTool.rewriteTsFile(tsFileResource, splitResources);
```

Comment:
The purpose here is to split and rewrite a tsFile only when it spans multiple partitions. When the tsFile is not partitioned across time, there is no need to split and rewrite it. Otherwise, we may need a function to determine whether the tsFile spans multiple time partitions. But that's exactly what the `getTimePartitionWithCheck()` function does, so I set the split logic here.

Reply:
I see, after we add a method to judgment whether it contains multiple partitions, then we do not need do the process logic in catch block.